recipes/pstrain/meta.yaml

```diff
@@ -0,0 +1,47 @@
+{% set version = "1.0.2" %}
+
+package:
+  name: pstrain
+  version: {{ version }}
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage("pstrain", max_pin="x") }}
+  entry_points:
+    - pstrain = PStrain_V30:main
+
+source:
+  url: https://github.com/wshuai294/PStrain/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 3a838a0d090509a08bc7b2e52aab832860542d0a8c00945f494e19d0240276f5
+
+requirements:
+  build:
+  host:
+  run:
+    - python >=3.7 
+    - metaphlan ==3
+    - pysam
+    - numpy 
+    - pulp
+    - samtools 
+    - bowtie2
+    - openjdk
+
+
+about:
+  home: https://github.com/wshuai294/PStrain
+  license: MIT
+  summary: An Iterative Microbial Strains Profiling Algorithm for Shotgun Metagenomic Sequencing Data
+
+test:
+  commands:
+    - pstrain --help
+
+extra:
+  skip-lints:
+    - should_be_noarch_generic
```

Comment:
any ideas why we can not have a `noarch: generic`.

Reply:
Hello, I have tested `noarch: generic` and `noarch: python`, but both of them failed. I think the reason might be that dependent packages like Metaphlan3 are not platform-independent.

Reply:
Hello, I have tested `noarch: generic` and it did not work. I wonder if it is necessary to add this statement. Thanks!

Reply:
How did it break ?

Reply:
I'd not expect to see such lint error with `noarch: generic`

Reply:
https://bioconda.github.io/contributor/linting.html#should-be-noarch-generic
I think your package contains both Java (*.jar) and Python code (*.py). I guess this is the reason why it fails ...

Reply:
That makes sense. So, could I just remove `noarch: generic` and skip the lint check?

Reply:
I don't have much experience with this. Let's see what more experienced devs will say!

Reply:
Okay. Thanks!